### PR TITLE
Feature/ak/batch job filelist 2

### PR DESCRIPTION
**Pull request by Akshith-Patil**
## Proposed changes

Had issues with the previous branch, hence created a new branch before the revert commit. 

Created new file management design for upload section
1. Added 'search' bar for upload section
2. New 'remove all' button to delete all uploaded files
3. Thumbnails for image and video files
4. Pagination for Batch Job upload section
5. Modal popup to view files in full-sized view

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

After uploading
After Upload

Clicking on the file thumbnail
Modal Popup

### CODE REVIEW

**Review by mt-empty — CHANGES_REQUESTED** [review]
When you get the error message `Something went wrong`, the local files are no longer available for viewing  

image

Don't fix it if it takes too long.

**Comment by Akshith-Patil** [comment]
> When you get the error message `Something went wrong, the local files are no longer available for viewing

That's strange, how did you get the `Something went wrong` error in the first place?

**Review by mt-empty — APPROVED** [review]

**Comment by idomo1** [comment]
If I have 2 pages of files, and remove all the files from the second page, I will be left with an empty list until I select page 1
image

**Comment by idomo1** [comment]
Is it an easy thing to add the controls to the bottom as well?

File: src/components/reportPage/UploadSection.jsx [hunk 1]
```diff
@@ -27,9 +28,26 @@ const UploadSection = (props) => {
...
+
+  // Search files
+  const [searchTerm, setSearchString] = useState("");
+
+  // Modal popup
+  const [modal, setModal] = useState(false);
+  const [fileSrc, setFileSrc] = useState();
+
   const removeFile = (file) => {
+    var file_index = file;
```
Comment (idomo1) [inline src/components/reportPage/UploadSection.jsx:45]: Should be `const` and should be in camelCase (`fileIndex`)

**Review by idomo1 — APPROVED** [review]

**Pull request merged**